server/src/common/dto/query/map-queries.dto.ts

```diff
@@ -133,3 +134,45 @@ export class MapCreditsGetQuery {
     @ExpandQueryDecorators(['user'])
     expand: string[];
 }
+
+export class MapRanksGetQuery extends PaginationQuery {
+    @ApiPropertyOptional({
+        name: 'playerID',
+        type: Number,
+        description: 'Steam ID of player to get rank for'
+    })
+    @IsInt()
+    @Type(() => Number)
+    @IsOptional()
+    playerID: number;
+
+    @ApiPropertyOptional({
+        name: 'playerIDs',
+        type: Number,
+        isArray: true,
+        description: 'List of steam IDs of players to get rank for'
+    })
+    @IsArray()
+    @Transform(({ value }) => value.map((v) => Number.parseInt(v)))
```

Comment:
I think you can just use `Type()` here since class-transformer will figure out you're transforming an array https://github.com/typestack/class-transformer#working-with-arrays

Reply:
But wait, this doesn't even make sense. This is on a query, which doesn't support arrays; it's just plaintext. So you want to pass a string and split it, like here (L48 user-queries.dto.ts)
```ts
@ApiPropertyOptional({
    name: 'userIDs',
    type: String,
    description: 'Filter by CSV list of Steam Community IDs',
    example: '123135674,7987347263,98312287631'
})
@Transform(({ value }) => value.split(','))
@IsSteamCommunityID({ each: true })
@IsOptional()
steamIDs: string[];
```

Reply:
queries do actually support arrays
`?userIDs=1234&userIDs=5678` should make `"userIDs": ["1234", "5678"]`